### PR TITLE
Add new route to add BGS notes

### PR DESCRIPTION
**Pull request by yoomlam**
## What was the problem?

Need BGS Notes to be added at the end of VRO processing.

- #1341

## How does this fix it?

Add new route to add BGS notes, after RFD is set.
Updated https://github.com/department-of-veterans-affairs/abd-vro/wiki/VRO-v2-Roadmap

### CODE REVIEW

**Comment by yoomlam** [comment]
[PR: End-to-end test for v2](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4554917586)

File: service/provider/src/main/java/gov/va/vro/service/provider/bgs/service/BgsApiClient.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      case EXAM_ORDER:
+        // Tested with VroV2Tests.testAutomatedClaimOrderExamNewClaim
+        log.warn("++++++++ EXAM_ORDER +++++++");
+        dto.veteranNotes.add(getArsdUploadedNote(getDocUploadedAt(mpo)));
+        dto.claimNotes.add(BgsClaimNotes.EXAM_REQUESTED_NOTE);
+        break;
+      case OFF_RAMP:
+        // Tested with VroV2Tests.testAutomatedClaimSufficiencyIsNull,
+        // which only covers SUFFICIENCY_UNDETERMINED
+        // Need to test the other 2 offrampError scenarios
```
Comment (yoomlam) [inline service/provider/src/main/java/gov/va/vro/service/provider/bgs/service/BgsApiClient.java:50]: Relying on unit test to cover the other 2 offrampError scenarios. @ioarash

File: shared/domain/src/main/java/gov/va/vro/model/bgs/BgsApiClientRequest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import lombok.RequiredArgsConstructor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Data
+@RequiredArgsConstructor
+public class BgsApiClientRequest {
+
+  final String vbmsClaimId;
```
Comment (hayleymdawson) [inline shared/domain/src/main/java/gov/va/vro/model/bgs/BgsApiClientRequest.java:13]: might want to comment what this ties to as i know you want to rename the other files to match this but they aren't there yet.

File: service/provider/src/main/java/gov/va/vro/service/provider/bgs/service/EntityNotFoundException.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package gov.va.vro.service.provider.bgs.service;
```
Comment (hayleymdawson) [inline service/provider/src/main/java/gov/va/vro/service/provider/bgs/service/EntityNotFoundException.java:1]: Does this belong in BGS or something higher as a more generic error?

File: service/provider/src/main/java/gov/va/vro/service/provider/bgs/service/BgsApiClient.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+import java.util.Optional;
+import java.util.UUID;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class BgsApiClient {
+  private final EvidenceSummaryDocumentRepository esDocRepository;
+
+  private int retryDelayBaseMills = 130_000;
```
Comment (hayleymdawson) [inline service/provider/src/main/java/gov/va/vro/service/provider/bgs/service/BgsApiClient.java:27]: that's a valid int format? (no complaints, just news to me)

File: service/provider/src/main/java/gov/va/vro/service/provider/bgs/service/BgsApiClient.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+  public BgsNotesCamelBody buildRequest(MasProcessingObject mpo) {
+    BgsNotesCamelBody body = new BgsNotesCamelBody(mpo, retryDelayBaseMills);
+    MasCompletionStatus completionStatus = MasCompletionStatus.of(mpo);
+    switch (completionStatus) {
+      case READY_FOR_DECISION:
+        // Tested with VroV2Tests.testAutomatedClaimFullPositiveIncrease
+        log.warn("++++++++ RFD +++++++");
```
Comment (hayleymdawson) [inline service/provider/src/main/java/gov/va/vro/service/provider/bgs/service/BgsApiClient.java:35]: warn over info? as it'll warn for everything.

File: service/provider/src/main/java/gov/va/vro/service/provider/camel/BgsApiClientRoutes.java [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    from(ADD_NOTES_RETRIES)
+        .doTry()
+        .log("microservice request: ${body}")
+        .process(requestBgsToAddNotesProcessor())
+        .id("requestBgsToAddNotes")
+        .doCatch(BgsApiClientException.class)
+        .setBody(simple("${body.incrementTryCount()}"))
+        .log(
+            "caught: ${exception.message}: tryCount=${body.tryCount} will retry in ${body.delayMillis} ms")
+        .delay(simple("${body.delayMillis}"))
```
Comment (hayleymdawson) [inline service/provider/src/main/java/gov/va/vro/service/provider/camel/BgsApiClientRoutes.java:68]: Clever. Given the trouble I have had with onException and any sort of retry with that code path I rather like this way of handling it.

File: service/provider/src/main/java/gov/va/vro/service/provider/camel/BgsApiClientRoutes.java [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+  public void configure() throws Exception {
+    var routeId = "add-bgs-notes-route";
+    configureAddNotesRoute(routeId);
+    configureRouteToBgsApiClientMicroservice();
+    configureRouteToSlackNotification();
+
+    String env = System.getenv("ENV");
+    log.info("ENV=" + env);
+    // TODO: workaround until a BGS Mock is implemented
+    if (!Arrays.asList(relevantEnvs).contains(env)) configureMockBgsApiMicroservice();
```
Comment (yoomlam) [inline service/provider/src/main/java/gov/va/vro/service/provider/camel/BgsApiClientRoutes.java:50]: @greene858 Here's the workaround

**Pull request merged**